### DISCUSSION TITLE
CoronaSpades - lacking documentation

### DISCUSSION THREAD

**Discussion by DarrenObbard** [ablab/spades#925] [Q&A] [answered]
### Description of bug

I cannot find enough information on how coronaspades makes use of the hmms to assemble viruses (from the users perspective, not the perspective of how it actually works)

Suppose I am not looking for just one family of viruses, but many. 

If I provide hmms for many different virus families simultaneously, will this make a mess? i.e., will it attempt to find scaffolds that have as many of those proteins as possible in them, even though they are from different families?

I see that by default coronaspades keeps all of the coronavirus hmms in one file. Is this a convenience, or does it imply that I should separate my virus protein hmms by family? 

### spades.log

This is not relevant

### params.txt

this is not relevant

### SPAdes version

Spades 3.15.4

### Operating System

Scientific linux

### Method of SPAdes installation

conda

### No errors reported in spades.log

- [X] Yes

**Comment by asl** [comment, marked as answer]
Hello

The description of how coronaSPAdes / rnaviralSPAdes uses HMMs is described in paper: https://pubmed.ncbi.nlm.nih.gov/34406356/

From the user perspective we suggest to split HMM set by family (similar to what is done for Coronaviridae HMMs). You can mix HMMs if you can be sure there are no cross-species hits.

**Comment by DarrenObbard** [comment]
Great!

Does that mean my specified directory can contain one hmm file per virus family, and each file will be used to separately guide the assembly, or do I have to run the code separately for each hmm/family?

Thanks,

**Reply by asl** [reply]
You need to run assembly separately. If multiple HMMs are in the same directory they will be treated together.

**Reply by DarrenObbard** [reply]
I wonder if that could become a feature request? (that each HMM file, or subdirectories) are used to separately guide the path? 

Otherwise its not really useful for virus discovery, since one would have to run all of the code as many times as their are virus families!

**Reply by asl** [reply]
Well, it does not make much sense as normally you're having many HMMs for certain viral protein families; having "single HMM per virus" does look strange. For example, the set of coronaviral HMMs contains 66 HMMs, some of them are quite specific (e.g. for accessory proteins of alphacoronaviruses or Spike protein of toroviruses) and some are very broad – e.g. the model for RdRP.

The set of HMMs should ideally cover the majority of viral proteins from end to end, so it should be broad, but specific. E.g. running over, say, whole RVDB-Prot-HMM would likely produce lots of chimeric results due to spurious matches.

PS: Note that there is no need to re-run everything from scratch - one can use the --assembly-graph option to augment cmdline with assembly graph and skip maybe 3/4 of "common" runtime.
